test-framework/core/src/main/java/org/keycloak/test/framework/realm/ManagedRealmCleanup.java

```diff
@@ -0,0 +1,49 @@
+package org.keycloak.test.framework.realm;
+
+import org.keycloak.admin.client.resource.RealmResource;
+import org.keycloak.representations.idm.RealmRepresentation;
+
+import java.util.LinkedList;
+import java.util.List;
+
+public class ManagedRealmCleanup {
+
+    private final List<RealmCleanup> cleanupTasks = new LinkedList<>();
+
+    public ManagedRealmCleanup add(RealmCleanup realmCleanup) {
+        this.cleanupTasks.add(realmCleanup);
+        return this;
+    }
+
+    public ManagedRealmCleanup deleteUsers() {
+        return add(r -> r.users().list().forEach(u -> r.users().delete(u.getId()).close()));
+    }
+
+    void resetToOriginalRepresentation(RealmRepresentation rep) {
+        if (cleanupTasks.stream().noneMatch(c -> c instanceof ResetRealm)) {
+            RealmRepresentation clone = RepresentationUtils.clone(rep);
+            cleanupTasks.add(new ResetRealm(clone));
+        }
+    }
+
+    void runCleanupTasks(RealmResource realm) {
```

Comment:
Same here: 
```suggestion
    public void runCleanupTasks(RealmResource realm) {
```

Reply:
Same, this should only be called by `ManagedRealm`